### DISCUSSION TITLE
high revalidation value for Incremental Static Regeneration, like:  "86400"

### DISCUSSION THREAD

**Discussion by stefanibus** [vercel/next.js#28962] [Help] [answered]
TL DR  : -- > 
Is it feasible to revalidate the data (delivered from my endpoints) only once a day with ISG? 

is there a limit to the value I can pass to the ISG key "revalidate" ? 
I did consult https://vercel.com/docs/edge-network/caching#stale-while-revalidate
 

LONG VERSION: 
I created a nextJS-application performing some crawling and scraping.
 
The next application will request  endpoints 
and those will perform rather data-heavy requests with a responsetime above 90 seconds.

I would prefer to only run those Endpoints   
ONE TIME PER  DAY. 

So  I neither want them to be requested  at build time  , 
nor do I aim for a serverside request on each request. 

Thus:  

I am wondering if it is advisable to use ISG  to acceive my goal? 

i would want to set the revalidation Value to  "86400 seconds" (which is = 1  day). 
  
I did not find a single example with a value higher than 60 seconds,
so I feel  I might be missing something here. 

Will Vercel be able to handle high revalidation values  ?  
The vercel-server would have to cound down those 86400 seconds  every day. 
And at the end of this timespan, 
the server would start counting the next time-span right away, 
until the next day has passed.

I guess the application will certainly see many days with zero visits.
No requests would be fired towards the endpoints on those days, 
as  no revalidation will get requested     by any user (by the server after the 86400 s delay).

**Comment by wilsonneto-swe** [comment]
Hi man!
Yes, of course, you can use a day as the time to revalidate, I use this approach in the "Policy and Terms" pages and thinks like that (I consume an endpoint, but only one time a day).

About the revalidation, you will need a request to start the revalidation, for example, if your site doesn't have any requests in one day, it won't revalidate the cache... 

And whats is the problem run the revalidate process on build time? (This is the normal approach)

The Vercel will love this kind of revalidation, it will consume less computer power and will just use static files / CDN all day, the best thing for them.

But we have a problem here, you need this revalidation to be exactly at a time in a day? Because your revalidation will occur only when you receive a request after the revalidation time... ok?

Another approach is to use full static content, and use Github Actions to schedule a new build and deploy process every day at the same time.

I hope it helps you.

**Comment by leerob** [comment, marked as answer]
I'm incredibly excited to share that Next.js 12.1 was just released, which includes **on-demand Incremental Static Regeneration**! You don't need to set a really high revalidation time anymore.

- [Learn about Next.js 12.1](https://nextjs.org/blog/next-12-1)
- [Watch a demo of on-demand ISR](https://youtu.be/BGexHR1tuOA)
- [View another demo and give us your feedback](https://on-demand-isr.vercel.app/)
- [Read the documentation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation-beta)
